File: Desenvolvimento/client/src/@fuse/types/models/endereco.ts
import { Municipio } from './municipio';
import { SetorEndereco } from './setor-endereco';
import { EmpresaEndereco } from './empresa-endereco';
import { BaseModel } from '../base-model';

export class Endereco extends BaseModel {
    constructor(
        public id: number,
        public cep?: string,
        public logradouro?: string,
        public complemento?: string,
        public bairro?: string,
        public numero?: string,
        public municipioId?: number,
        public municipio?: Municipio, 
        public empresaEnderecos?: Array<EmpresaEndereco>,
        public setorEnderecos?: Array<SetorEndereco>
        ) { super(); }
}

File: Desenvolvimento/server/src/ApplicationCore/Entities/Endereco.cs
using ApplicationCore.Interfaces.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace ApplicationCore.Entities
{
    public class Endereco : IBaseEntity<long>
    {
        public long Id { get; set; }
        public string CEP { get; set; }
        public string Logradouro { get; set; }
        public string Complemento { get; set; }
        public string Bairro { get; set; }
        public string Numero { get; set; }

        public long? MunicipioId { get; set; }
        public virtual Municipio Municipio  { get; set; }

        public virtual ICollection<EmpresaEndereco> EmpresaEnderecos { get; set; }
        public virtual ICollection<SetorEndereco> SetorEnderecos { get; set; }
    }
}
